src/Main.java

```diff
@@ -1,307 +1,466 @@
-                listarSintomas(scanner, listaSintomas, input);
-                break;
-            case "3":
-                realizarAnamnese(listaSintomas, scanner, input);
-                break;
-            case "4":
-                System.out.println("\nObrigado por utilizar PUC Anamnese!");
-                return false;
-            default:
-                System.out.println("\nOpção Inválida");
-                break;
-        }
-        return true;
+    return keepProcess;
+  }
+
+  /**
+   * Listar Doenças.
+   * Este método é responsável por listar as doenças reconhecidas pelo sistema
+   */
+  private static void listarDoencas() {
+    System.out.println("\nAs doenças reconhecidas pelo sistema são:\n");
+
+    for(String doenca : Main.DOENCAS) {
+      System.out.println(
+        "- " + doenca
+      );
     }
+    System.out.println("\n--");
+    System.out.println("\nVamos continuar?\n");
+  }
+
+  /**
+   * Listar Sintomas de uma doença
+   * Este método é responsável pelo menu de listar os sintomas das doenças
+   * 
+   * @param Scanner scanner - Objeto que possui suporte de resgatar entrada do
+   *                          usuário
+   */
+  private static void listarSintomas(Scanner scanner) {
+    System.out.println(
+      "\nVamos listar os possíveis sintomas reconhecidos. Para isso, selecione uma doença:"
+    );
+
+    String input = "";
+
+    do {
+      System.out.println("\n--");
+
+      int i;
+      for(i = 0; i < Main.DOENCAS.length; i++) {
+        System.out.println(
+          (i+1) + ") " + Main.DOENCAS[i]
+        );
+      }
+      System.out.println("--");
+      System.out.println("V) Voltar");
+
+      System.out.println("--\n");
+      System.out.print(
+        "Escolha uma doença para para listar os sitomas: "
+      );
+      input = scanner.nextLine().toUpperCase();
+
+      if (input.equals("V")) {
+        // Define o loop até aqui.
+        continue;
+      }
 
-    /**
-     * Opção 1 - Listar Doenças.
-     *
-     * Este método é responsável por listar as doenças reconhecidas pelo sistema
-     *
-     * @param scanner
-     * @param input
-     */
-    private static void listarDoencas() {
-        System.out.println("\nAs doenças reconhecidas pelo sistema são:\n");
-        System.out.println("- Influenza (H1N1)\n" + "- Pneumonia\n" + "- Corona Virus (Covid-19)");
-        System.out.println("\n--");
-        System.out.println("\nVamos continuar?\n");
+      Integer opcaoInput = 0;
+
+      try {
+        // Conversão do input inforado pelo usuário
+        opcaoInput = Integer.parseInt(input);
+      } catch(NumberFormatException e) {
+        System.out.println("Informe um número válido.");
+        continue;
+      }
+
+      if (opcaoInput > Main.DOENCAS.length) {
+        // Se o indice informado não está dentro do limite de doenças suportadas,
+        // volta ao início perguntando novamente.
+        System.out.println("\nOpção inválida.");
+        continue;
+      }
```

Comment:
A verificação da linha `280` permite que o usuário digite 0 ou um número negativo, uma vez que está só verifica se o input é menor que `Main.DOENCAS.length`. Isso ocasiona um `ArrayIndexOutOfBoundsException` na linha 
`391` - `System.out.println("> Sintoma: " + opcaoInput + " - " + Main.SINTOMAS[index]);` 
onde o sistema tentará acessar um range inválido do array `Main.SINTOMAS`.
Para evitar isso sugiro a seguinte correção:
- Ao invés de verificar o limite máximo do input em relação ao tamanho do array `Main.DOENCAS` delimitar um intervalo entre 1 (inclusive) e `Main.DOENCAS.length`.

```suggestion
      if (opcaoInput < 1 || opcaoInput > Main.DOENCAS.length) {
        // Se o indice informado não está dentro do limite de doenças suportadas,
        // volta ao início perguntando novamente.
        System.out.println("\nOpção inválida.");
        continue;
      }
```